web/static/internal/apps/restock/restock.js

```diff
@@ -33,30 +33,52 @@ document.onkeydown = ({ key }) => {
   }
 };
 
+function valid_upc(upc) {
```

Comment:
Definitely add a comment to this function that describes the properties we're checking and a link to some sort of spec or source.

Reply:
Oh also include a small explanation of why we're validating the UPC's (namely the scanners sometimes screw up)